cli/internal/cloudcmd/upgrade.go

```diff
@@ -180,24 +203,62 @@ func (u *kubeImageUpdater) getCurrent(ctx context.Context, name string) (*unstru
 }
 
 // update updates the image definition.
-func (u *kubeImageUpdater) update(ctx context.Context, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+func (u *dynamicClient) update(ctx context.Context, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
 	return u.client.Resource(schema.GroupVersionResource{
 		Group:    "update.edgeless.systems",
 		Version:  "v1alpha1",
 		Resource: "nodeimages",
 	}).Update(ctx, obj, metav1.UpdateOptions{})
 }
 
-type kubeMeasurementsUpdater struct {
+type stableClient struct {
 	client kubernetes.Interface
 }
 
 // getCurrent returns the cluster's expected measurements.
-func (u *kubeMeasurementsUpdater) getCurrent(ctx context.Context, name string) (*corev1.ConfigMap, error) {
+func (u *stableClient) getCurrent(ctx context.Context, name string) (*corev1.ConfigMap, error) {
 	return u.client.CoreV1().ConfigMaps(constants.ConstellationNamespace).Get(ctx, name, metav1.GetOptions{})
 }
 
 // update updates the cluster's expected measurements in Kubernetes.
-func (u *kubeMeasurementsUpdater) update(ctx context.Context, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+func (u *stableClient) update(ctx context.Context, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
 	return u.client.CoreV1().ConfigMaps(constants.ConstellationNamespace).Update(ctx, configMap, metav1.UpdateOptions{})
 }
+
+func (u *stableClient) kubernetesVersion() (string, error) {
+	serverVersion, err := u.client.Discovery().ServerVersion()
+	if err != nil {
+		return "", err
+	}
+	return serverVersion.GitVersion, nil
+}
+
+type helmInterface interface {
+	currentVersion(release string) (string, error)
+}
+
+type helmClient struct {
+	config *action.Configuration
+}
+
+func (c *helmClient) currentVersion(release string) (string, error) {
+	client := action.NewList(c.config)
```

Comment:
Did you check if setting the namespace or `AllNamespaces` on the action is required?